test/built-ins/Array/prototype/fill/fill-values.js

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.1.3.6
+description: >
+  Fills all the elements from a start index to an end index with `value`.
+info: >
+  22.1.3.6 Array.prototype.fill (value [ , start [ , end ] ] )
+
+  ...
+  7. If relativeStart < 0, let k be max((len + relativeStart),0); else let k be
+  min(relativeStart, len).
+  8. If end is undefined, let relativeEnd be len; else let relativeEnd be
+  ToInteger(end).
+  9. ReturnIfAbrupt(relativeEnd).
+  10. If relativeEnd < 0, let final be max((len + relativeEnd),0); else let
+  final be min(relativeEnd, len).
+  11. Repeat, while k < final
+    a. Let Pk be ToString(k).
+    b. Let setStatus be Set(O, Pk, value, true).
+    c. ReturnIfAbrupt(setStatus).
+    d. Increase k by 1.
+  12. Return O.
```

Comment:
relativeStart and relativeEnd is a first good separation. I'll split this way and see if it's good to go further and split again.

They're together as they came this way from the old tests. I had the same impression but I wanted to check first it this would smell in a code review, as it just did.